docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+    // NOTE: this field is part of the Cluster API contract, and it is used to orchestrate initial Machine provisioning.
+    // The value of this field is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's BootstrapSecret.
+    // +optional
+    BootstrapSecretCreated bool `json:"bootstrapSecretCreated"`
+    
+    // InfrastructureProvisioned is true when the infrastructure provider reports that the Machine's infrastructure is fully provisioned.
+    // NOTE: this field is part of the Cluster API contract, and it is used to orchestrate initial Machine  provisioning.
+    // The value of this field is never updated after provisioning is completed.
+    // Use conditions to monitor the operational state of the Machine's infrastructure.
+    // +optional
+    InfrastructureProvisioned bool `json:"infrastructureProvisioned"`
+}
+```
+
+| v1beta1 (current)             | v1beta2 (tentative Q1 2025)                              | v1beta1 removal (tentative Q1 2026)        |
+|-------------------------------|----------------------------------------------------------|--------------------------------------------|
+|                               | `Initialization` (new)                                   | `Initialization`                           |
+| `BootstrapReady`              | `Initialization.BootstrapSecretCreated` (renamed)        | `Initialization.BootstrapSecretCreated`    |
+| `InfrastructureReady`         | `Initialization.InfrastructureProvisioned` (renamed)     | `Initialization.InfrastructureProvisioned` |
+|                               | `BackCompatibilty` (new)                                 | (removed)                                  |
+| `Phase` (deprecated)          | `BackCompatibilty.Phase` (renamed) (deprecated)          | (removed)                                  |
+| `LastUpdated` (deprecated)    | `BackCompatibilty.LastUpdated` (renamed) (deprecated)    | (removed)                                  |
+| `FailureReason` (deprecated)  | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                                  |
+| `FailureMessage` (deprecated) | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                                  |
+| `Conditions`                  | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                                  |
+| `ExprimentalConditions` (new) | `Conditions` (renamed)                                   | `Conditions`                               |
+| other fields...               | other fields...                                          | other fields...                            |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
```

Comment:
Why do we need to keep this in a sub-struct in v1beta2?

Reply:
I took this idea from https://kubernetes.io/docs/reference/using-api/deprecation-policy/#fields-of-rest-resources.

This was sort of turning point in preparing this design proposal, because the fact of having two fields (one representing the end state, one for back compatibility / down conversions) is what allowed me to think about an ideal target state.

Reply:
I agree with the interpretation there, and I think I can answer my own question, yes, this would be set by the controller. SGTM on this concept at least, I think I had a question about the name but we can discuss on the other thread

Reply:
Agree. I think we have to ensure that folks using v1beta1 get conditions with the same semantic as today. This is not possible with down conversion from v1beta2 as some of the new conditions have a different semantic

Reply:
Resolving this comment since there is agreement.
Also, there are other threads discussion the naming for the back compatibility field, so they willl be still in the radar